src/components/BlogList.jsx

```diff
@@ -1,23 +1,53 @@
 import BlogPost from "./BlogPost";
 import Pagination from "./Pagination";
-import React from "react";
+import React, { useState } from "react";
 import blogs from "../data/blogs.json";
 
 const PAGE_SIZES = [15, 25, 50, 100];
 
 function BlogList() {
-  const currentPaginationData = blogs.posts.slice(0, 15);
+  const [rowsPerPage, setRowsPerPage] = useState(15);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [arrayStart, setArrayStart] = useState(0);
 
-  const updateRowsPerPage = () => {};
-  const updatePage = () => {};
+  let arrayEnd = arrayStart + rowsPerPage;
+  let maxPageNumber = Math.ceil(blogs.posts.length / rowsPerPage);
+
+  const currentPaginationData = blogs.posts.slice(arrayStart, arrayEnd);
```

Comment:
You can avoid recalculating (slicing) on every render by placing the blog posts data in a state.